frontend/tests/playwright/msw/request-handler.ts

```diff
@@ -0,0 +1,233 @@
+import { isRecord } from '@ifixit/helpers';
+import {
+   GraphQLHandler,
+   RestHandler,
+   RequestHandler,
+   DefaultBodyType,
+   GraphQLContext,
+   GraphQLRequest,
+   GraphQLVariables,
+   PathParams,
+   ResponseResolver,
+   RestContext,
+   RestRequest,
+} from 'msw';
+
+/**
+ * These methods are a subset of the GraphQL operational methods that are
+ * supported by MSW's `graphql` handler. There are of course other methods
+ * `MSW` supports, but we are only interested in these two for now.
+ *
+ * @see https://mswjs.io/docs/api/graphql#operation-kind
+ */
+export type GraphQLMethods = 'mutation' | 'query';
+/**
+ * This is the type of the resolver that is used by the `graphql` handler.
+ * The resolver is a function that accepts a captured request and may return
+ * a mocked response.
+ * @see https://mswjs.io/docs/basics/response-resolver
+ */
+export type GraphQLResolver = ResponseResolver<
+   GraphQLRequest<GraphQLVariables>,
+   GraphQLContext<Record<string, any>>
+>;
+/**
+ * These methods are a subset of the REST methods that are
+ * supported by MSW's `rest` handler - _with the exception of
+ * `options`_.
+ *
+ * @see https://mswjs.io/docs/api/rest#methods
+ */
+export type RestMethods = 'all' | 'get' | 'post' | 'put' | 'patch' | 'delete';
+/**
+ * This is the type of the resolver that is used by the `rest` handler.
+ * The resolver is a function that accepts a captured request and may return
+ * a mocked response.
+ * @see https://mswjs.io/docs/basics/response-resolver
+ */
+export type RestResolver = ResponseResolver<
+   RestRequest<DefaultBodyType, PathParams<string>>,
+   RestContext,
+   DefaultBodyType
+>;
+
+export type RequestInfo = {
+   endpoint: string;
+   method: GraphQLMethods | RestMethods;
+};
```

Comment:
I think it'll be easier to work with to have two copies of this: one for GraphQL and one for REST. The fact that `endpoint` is the same type and would be duplicated seems relatively unimportant to me. If you disagree, you can always define a `BaseRequestInfo` with just `{ endpoint: string }` in it.

Reply:
If we make two respective request info types for each API type, should we remove the overloading for `create()`?